src/hotspot/os/windows/os_windows.cpp

```diff
@@ -5315,6 +5315,22 @@ void os::funlockfile(FILE* fp) {
   _unlock_file(fp);
 }
 
+char* os::realpath(const char* filename, char* outbuf, size_t outbuflen) {
+
+  if (filename == nullptr || outbuf == nullptr || outbuflen < 1) {
+    assert(false, "os::realpath: invalid arguments.");
+    errno = EINVAL;
+    return nullptr;
+  }
+
+  char* result = nullptr;
+  ALLOW_C_FUNCTION(::_fullpath, result = ::_fullpath(outbuf, filename, outbuflen);)
+  if (result == nullptr) {
+    errno = ENAMETOOLONG;
+  }
```

Comment:
This is a bit of an assumption. What if the name "includes a drive letter that isn't valid or can't be found"? Unfortunately Windows doesn't specify any further details beyond returning null.

Reply:
I probably cleaned up this code too much, and should've left it more like the Posix implementation.  
What I used to have would do one call (with buffer NULL) to get the real full path, then copy it if it fit or ENAMETOOLONG.  In attempting to speed up the code I changed the semantics.  I will change this code back to my previous implementation

Reply:
changed code to test for ENAMETOOLONG.